activejob/test/models/custom_type.rb

```diff
@@ -0,0 +1,16 @@
+class CustomType
+  # Enable custom serialization for this class
+  include ActiveJob::Serialization
+  # Some attributes to serialize
+  attr_reader :foo, :bar
+
+  def initialize(v1, v2)
+    @foo = v1
+    @bar = v2
+  end
+
+  # equality method used in tests
+  def ==(obj)
```

Comment:
Do you want to keep a separated file or the whole structure inlined in test case? Something like

```
test 'should use custom serialization when implemented' do
  CustomType = Struct.new('CustomType', :foo, :bar)  do
    include ActiveJob::Serialization
  end
  value = CustomType.new(rand(1000), rand(1000))
  assert_arguments_roundtrip [value]
end
```